examples/tutorials/advanced/legends.py

```diff
@@ -0,0 +1,138 @@
+"""
+Creating legends
+================
+
+The :meth:`pygmt.Figure.legend` method creates legends, whereby auto-legends as
+well as manually created legends are supported.
+
+Users can adjust the position of the legend and add a box around the legend.
+"""
+
+# %%
+import io
+
+import pygmt
+
+# %%
+# Create an auto-legend
+# ---------------------
+#
+# For auto-legends, the ``label`` parameter of :meth:`pygmt.Figure.plot` has to
+# be specified to state the desired text in the legend entry (white spaces are
+# allowed). Optionally, to adjust the legend, users can append different modifiers
+# to the string passed to ``label`. A list of all available modifiers can be
+# found at :gmt-docs:`gmt.html#l-full`. To create a multiple-column legend **+N**
+# is used with the desired number of columns; see also gallery example
+# https://www.pygmt.org/dev/gallery/embellishments/legend.html.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=-3, style="c0.25c", fill="orange", label="orange circle")
+fig.legend()
+
+fig.show()
+
+
+# %%
+# Adjust the position
+# -------------------
+# Use the ``position`` parameter to adjust the position of the legend.
+# j within, J outside of bounding box
+# also adjust ``width`` via **+w** modifier
+# The default of ``box`` is changed, i.e. no box plotted anymore.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=-3, style="c0.25c", fill="orange", label="orange circle")
+fig.legend(position="jMC")
```

Comment:
It makes no sense to put legends in the middle center of a plot. Maybe just put it on the top-left to match realistic use cases?

Reply:
Yeah, true - makes for sense to move the legend instead of the data point 😅; updated in commit https://github.com/GenericMappingTools/pygmt/pull/3594/commits/46e8340baed882e30b5fcc8ded40666ac6987d57.